### DISCUSSION TITLE
How to get compiled extensions instead of compiling them on the spot?

### DISCUSSION THREAD

**Discussion by andersonpem** [mlocati/docker-php-extension-installer#891] [Q&A]
Hi. Sorry if this is a dumb question. I have a pipeline that requires building an image every time a commit is pushed.
Spending compute time compiling C code on every push is a waste of computing power and of time.

I'm preparing an image with these things pre-built to pull from. But is it possible to have compiled sources of these extensions? :)

Thank you for the awesome software.

Andy.

**Comment by mlocati** [comment]
In order to make PHP extensions work, having their compiled (`.so`) files is not enough. We also have to install some system (apt/apk) packages, and some of them also require some manual configuration.

If possible, prepare a docker image with the extensions you need, and start from it instead of using the default php docker images.

**Reply by zdev-code** [reply]
Why not just "apt-get install -y php-gd"? Otherwise there's an hour long build time to update the packages...

**Reply by mlocati** [reply]
> Why not just "apt-get install -y php-gd"? Otherwise there's an hour long build time to update the packages...

@akzov Because installing the `php-gd` APT package would also install the php provided by APT: you'll have two PHP versions installed (the one provided by APT and the one already installed in the Docker image).
And about the GD PHP extension: you can reduce its compilation time by using Debian 12+ or Alpine 3.15+; for older versions you can specify set the `IPE_GD_WITHOUTAVIF` environment variable to `1` (provided that you don't need AVIF support).
